osu.Game/Overlays/Mods/ModButton.cs

```diff
@@ -147,18 +147,21 @@ public Mod Mod
 
         public virtual Mod SelectedMod => Mods.ElementAtOrDefault(selectedIndex);
 
-        protected override bool OnMouseDown(InputState state, MouseDownEventArgs args)
+        protected override bool OnMouseUp(InputState state, MouseUpEventArgs args)
         {
-            switch (args.Button)
+            // only trigger the event if we are inside the area of the button
+            if (Contains(ToScreenSpace(state.Mouse.Position - Position)))
```

Comment:
Wait it does trigger the `OnMouseUp` of the mod button if you release while not on the button?